src/extension.ts

```diff
@@ -1,52 +1,134 @@
+  } 
+
+  // Add scores
+  scores.push(scoreData);
+
+  // If the file does not exist, a new file is created, and if it exists, the contents are updated.
+  fs.writeFileSync(scoresPath, JSON.stringify(scores, null, 2)); // Add formatting
+}
+
+/**
+ * message handler for a webview panel.
+ * This function listense for messages sent from the webview and performs actions based on the message content.
+ * @param panel 
+ * @param context 
+ * @param gameId 
+ */
+function handlePanelMessages(panel: vscode.WebviewPanel, context: vscode.ExtensionContext, gameId: string){
+  panel.webview.onDidReceiveMessage(
+    message => {
+      if (message.command === 'sendScore') {
+        const score = message.score;
+        const player = message.player;
+        console.log(score);
+        saveScore(context, gameId, {player, score});
+      }
+    },
+    undefined,
+    context.subscriptions
+  );
+
+}
+
 export function activate(context: vscode.ExtensionContext) {
   // GameProvider 인스턴스 생성
   const gameProvider = new GameProvider();
   // 'gameExplorer' 뷰 컨테이너에 대한 TreeDataProvider로 gameProvider를 등록
-  vscode.window.registerTreeDataProvider('gameExplorer', gameProvider);
-  // 인트로 영상 보기
-  createIntroPanel(context);
+  vscode.window.registerTreeDataProvider("gameExplorer", gameProvider);
   // 'gameExtension.openGame' 커맨드를 등록하고, 실행될 때 createGameWebview 함수를 호출
   const openGamePanel = (context: vscode.ExtensionContext, game: MetaData) => {
+    const scoreBoardProvider = new ScoreBoardProvider(context, game.id);
+    vscode.window.registerTreeDataProvider('scoreList', scoreBoardProvider);
+
     if (openWebviews.has(game.id)) {
       const webViewPanel = openWebviews.get(game.id);
       webViewPanel?.reveal();
     } else {
-      const panel = createWebviewPanel(context, game);
+      const panel = vscode.window.createWebviewPanel(
+        game.id, // 웹뷰의 유형
+        game.title, // 사용자에게 표시될 패널 제목
+        vscode.ViewColumn.Active, // 웹뷰 패널을 표시할 에디터 열
+        {
+          retainContextWhenHidden: true,
+          enableScripts: true, // 스크립트 활성화
+          localResourceRoots: [
+            vscode.Uri.file(
+              vscode.Uri.joinPath(context.extensionUri, "src/game", game.id)
+                .fsPath
+            ),
+            vscode.Uri.file(
+              vscode.Uri.joinPath(
+                context.extensionUri,
+                "resource/game",
+                game.id
+              ).fsPath
+            ),
+          ],
+        }
+      );
       panel.webview.html = getHtmlContent(context, panel.webview, game);
 
+      handlePanelMessages(panel, context, game.id);
       openWebviews.set(game.id, panel);
 
       panel.onDidDispose(() => openWebviews.delete(game.id));
+
+
     }
   };
 
   let openGameCommand = vscode.commands.registerCommand(
-    'vsc-gameboy.openGame',
-    gameId => {
-      let game = metaData.find(game => game.id === gameId);
+    "vsc-gameboy.openGame",
```

Comment:
We made a prettierrc.
I think your formatter seems not synchronized with our new prettier format.
pull main branch first and check it out!